### PR TITLE
add modeq implementation

### LINKED ISSUE #96: operator %= Implementation
Currently the bitwise not operator `%=` isn't implemented in pocketlang, and evaluating an expression with it causes a syntax error. 
The expression `a %= b` will be compiled as `a = a % b` and we already have the functionality for that. Adding the tokens for the operators is sufficient.

## How to implement
- I've made a walkthrough on how to implement `&=`, in #69 use it for a reference.
- #70 is another example for `|=` and `^=` which is done by #71

## Finally
Don't forget to add some tests (at test/lang/basics.pk).

### PR DESCRIPTION
**Pull request by Takashiidobe**
Resolves #96. 

I couldn't figure out how to add a 0..1000 range test for modeq.

### CODE REVIEW

File: src/pk_compiler.c [hunk 1]
```diff
@@ -1152,6 +1155,7 @@ GrammarRule rules[] = {  // Prefix       Infix             Infix Precedence
   /* TK_ANDEQ      */   NO_RULE,
   /* TK_OREQ       */   NO_RULE,
   /* TK_XOREQ      */   NO_RULE,
+  /* TK_MODEQ      */   NO_RULE,
```
Comment (ThakeeNathees) [inline src/pk_compiler.c]: The order of the tokens is very important here, `TK_MODEQ` comes after `TK_DIVEQ` in the `Tokens` so it should also match here.

**Review by ThakeeNathees — CHANGES_REQUESTED** [review]
Thank you for taking your time and effort to make this PR.

Everything looks great, I have a tiny change request, left in the review below. Once you changed the order it can be merged.

File: src/pk_compiler.c (hunk 1 above)
Comment (Takashiidobe) [reply src/pk_compiler.c]: Makes sense to me, change has been made.

File: src/pk_compiler.c [hunk 2]
```diff
@@ -1853,6 +1857,7 @@ static void emitAssignment(Compiler* compiler, TokenType assignment) {
     case TK_ANDEQ: emitOpcode(compiler, OP_BIT_AND); break;
     case TK_OREQ: emitOpcode(compiler, OP_BIT_OR); break;
     case TK_XOREQ: emitOpcode(compiler, OP_BIT_XOR); break;
+    case TK_MODEQ: emitOpcode(compiler, OP_MOD); break;
```
Comment (ThakeeNathees) [inline src/pk_compiler.c:1860]: Here is the order is wrong, but It's not that a big deal, I'll merge it and change myself.

**Review by ThakeeNathees — APPROVED** [review]

**Comment by ThakeeNathees** [comment]
Thanks!! We're glad to have you here 🎉

**Pull request merged**